packages/datadog-plugin-grpc/src/client.js

```diff
@@ -180,7 +180,7 @@ function ensureMetadata (client, args) {
     normalized.push(new client._datadog.grpc.Metadata())
   }
 
-  for (let i = 1; i < args.length; i++) {
+  for (let i = typeof args[1] == 'undefined' ? 2 : 1; i < args.length; i++) {
```

Comment:
Isn't this covered by the above condition that checks for `!args[1]`?

Reply:
It is, but the broken case here is when the first two arguments are something like `[request, undefined, callback]`. In that case, we want the normalized args to be `[request, metadata, callback]`. The `!args[1]` check correctly identifies that we have an undefined as the second arg, so we need to push a new metadata object in the normalized args. However, it does a _push_ instead of replacing it in-place, so the `undefined` remains, and then we have an incorrect number of arguments.

Reply:
Gotcha. I think this would be easier to understand with an `else` case above to push the existing metadata instead, and then always start looping from `2`.